src/NzbDrone.Core/Notifications/Trakt/TraktService.cs

```diff
@@ -0,0 +1,269 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Http;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.MediaFiles.MediaInfo;
+using NzbDrone.Core.Tv;
+using NzbDrone.Core.Notifications.Trakt.Resource;
+using NzbDrone.Core.Qualities;
+using NzbDrone.Core.MetadataSource.SkyHook.Resource;
+using NzbDrone.Core.Indexers.HDBits;
+using NzbDrone.Core.IndexerSearch;
+
+namespace NzbDrone.Core.Notifications.Trakt
+{
+    public interface ITraktService
+    {
+        HttpRequest GetOAuthRequest(string callbackUrl);
+        TraktAuthRefreshResource RefreshAuthToken(string refreshToken);
+        void AddEpisodeToCollection(TraktSettings settings, Series series, EpisodeFile episodeFile);
+        string GetUserName(string accessToken);
+        ValidationFailure Test(TraktSettings settings);
+    }
+
+    public class TraktService : ITraktService
+    {
+        private readonly ITraktProxy _proxy;
+        private readonly Logger _logger;
+
+        public TraktService(ITraktProxy proxy,
+                           Logger logger)
+        {
+            _proxy = proxy;
+            _logger = logger;
+        }
+
+        public string GetUserName(string accessToken)
+        {
+            return _proxy.GetUserName(accessToken);
+        }
+
+        public HttpRequest GetOAuthRequest(string callbackUrl)
+        {
+            return _proxy.GetOAuthRequest(callbackUrl);
+        }
+
+        public TraktAuthRefreshResource RefreshAuthToken(string refreshToken)
+        {
+            return _proxy.RefreshAuthToken(refreshToken);
+        }
+
+        public ValidationFailure Test(TraktSettings settings)
+        {
+            try
+            {
+                GetUserName(settings.AccessToken);
+                return null;
```

Comment:
Nit: New line before return (applies to other cases below as well).

Reply:
no problem to fix.